docs/docs/security-in-gatsby.md

```diff
@@ -1,9 +1,95 @@
 ---
 title: Security in Gatsby
-issue: https://github.com/gatsbyjs/gatsby/issues/13305
 ---
 
-This is a stub. Help our community expand it.
+By being a modern static site generator, Gatsby inherits several security principles that comes with this approach. However, even with this, your users could be exposed to malicious attacks. In this page, you will se what they are and how you could prevent them.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Cross-Site Scripting (XSS)
+
+Cross-Site Scripting is a type of attack that injects a script or a unexpected link to another site into the client side of the application.
+
+JSX elements automatically escapes HTML tags by design. So, when you try to inject them as the example below:
+
+```js
+// highlight-next-line
+const username = `<script src='https://path/to/badness.js'></script>`
+
+const User = () => <p> hello {username}</p>
+```
+
+it will render `hello <script src='https://path/to/badness.js'></script>` as a string inside the p tag.
+
+In other hand, there could be some fields in your application that you will need to render the inner HTML tags, such as a content field in a blog or a comment in a comments section, that are built in rich-text editors.
+
+That's when you expose your application to XSS attacks, since the only way to render these HTML tags in React is by using the `dangerouslySetInnerHTML` prop:
```

Comment:
This isn't entirely true. there are several other packages that do this. [html-to-react](https://www.npmjs.com/package/html-to-react) and [html-react-parser](https://www.npmjs.com/package/html-react-parser) are the two I've used. I think these both sanitize on the client side, but not on SSR (but don't quote me on that...). They offer a lot of additional features too but those are not inherently related to security. I'd include this information some how so folks no it's an option...I'll leave it to you exactly how this should be done.

Reply:
You are right. There are alternatives to `dangerouslySetInnerHTML` but they don't sanitize HTML. see https://github.com/remarkablemark/html-react-parser/issues/94 and https://github.com/aknuds1/html-to-react/issues/79.

I reformulate the phrase to mention the html-to-react package, what do you think?

Reply:
I'd at least mention them.  I'm fairly sure html-react-parser doesn't do script tags on the client side (I ran into this on a project). one could also use the custom parsing functionality in these to sanitize html

Reply:
I mentioned html-react-parser! Please check again and see that if it is good.